vexim/sitepasswordsubmit.php

```diff
@@ -11,13 +11,18 @@
     $success = $sth->execute(array(':crypt'=>$cryptedpassword));
     if ($success) {
       $_SESSION['crypt'] = $cryptedpassword;
+      if(isset($_SESSION['pw_enforce'])) unset($_SESSION['pw_enforce']);
       header ("Location: site.php?sitepass=success");
       die;
     } else {
       header ("Location: site.php?sitepass=fail");
       die;
     }
   } else {
+    if(isset($_SESSION['pw_enforce'])) {
+      header ("Location: sitepassword.php?badpass=siteadmin");
+      die;
+    }
     header ("Location: site.php?badpass=siteadmin");
```

Comment:
If you change your mind, you can click on "Manage domains" and you're back to siteadmin's main view. If you however have entered two different passwords, or an insufficient one twice, you probably did this by mistake, and not on purpose just to get out of the "change password" window. So IMO returning the user back to the change password page is perhaps the sanest thing to do anyway.